src/notation/internal/notationactioncontroller.cpp

```diff
@@ -954,22 +946,20 @@ void NotationActionController::addFret(int fretIndex)
 
 void NotationActionController::undo()
 {
-    auto notation = currentNotation();
-    if (!notation) {
+    auto interaction = currentNotationInteraction();
+    if (!interaction) {
         return;
     }
-
-    notation->undoStack()->undo();
+    interaction->undo();
```

Comment:
Needed to move this to the next layer down because the EditData structure is needed for undo/redo to work properly when editing text elements